examples/hikari_tanjun_basic_queue/music.py

```diff
@@ -0,0 +1,351 @@
+
+
+async def _join_voice(ctx: tanjun.abc.Context, lavalink: lavasnek_rs.Lavalink) -> int:
+    """Joins your voice channel."""
+    assert ctx.guild_id is not None
+
+    if ctx.client.cache and ctx.client.shards:
+        # Get the users voice state if they didnt pass a channel
+        if not (voice_state := ctx.client.cache.get_voice_state(ctx.guild_id, ctx.author)):
+            await ctx.respond("Please connect to a voice channel.")
+            return 1
+
+        # Make sure our user is cached
+        elif own_user := ctx.client.shards.get_me():
+            # Check if there is a voice channel for the author
+            if voice_state.channel_id and own_user.id in (
+                # Check if we are in that channel
+                ctx.client.cache.get_voice_states_view_for_channel(
+                    ctx.guild_id,
+                    voice_state.channel_id,
+                )
+            ):
+                # We are in the channel already
+                await ctx.respond(f"I am already connected to <#{voice_state.channel_id}>.")
+                return 1
+
+        # Bot joins voice
+        await ctx.client.shards.update_voice_state(
+            ctx.guild_id, voice_state.channel_id, self_deaf=True
+        )
+        # Lavasnek waits for the data on the event
+        conn = await lavalink.wait_for_full_connection_info_insert(ctx.guild_id)
+        # Lavasnek tells lavalink to connect
+        await lavalink.create_session(conn)
+
+        await ctx.respond(f"Connected to <#{voice_state.channel_id}>")
+        return 0
+
+    await ctx.respond("Unable to join voice. The cache is disabled or shards are down.")
+    return 1
+
+
+@music.with_slash_command
+@tanjun.with_str_slash_option("song", "The title or youtube link of the song you want to play.")
+@tanjun.as_slash_command("play", "Play a song, or add it to the queue.")
+async def play_as_slash(
+    ctx: tanjun.abc.SlashContext,
+    song: str,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    await _play_track(ctx, song, lavalink)
+
+
+@music.with_message_command
+@tanjun.with_greedy_argument("song")  # Set song to be greedy
+@tanjun.with_parser  # Add an argument parser to the command
+@tanjun.as_message_command("play")
+async def play_as_message(
+    ctx: tanjun.abc.MessageContext,
+    song: str,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    """Play a song, or add it to the queue."""
+    await _play_track(ctx, song, lavalink)
+
+
+async def _play_track(ctx: tanjun.abc.Context, song: str, lavalink: lavasnek_rs.Lavalink) -> None:
+    """Attempts to play the song from youtube."""
+    assert ctx.guild_id is not None
+
+    # Check if we are connected to voice
+    conn = await lavalink.get_guild_gateway_connection_info(ctx.guild_id)
+
+    if not conn:
+        # Join the users voice channel if we are not already connected
+        if await _join_voice(ctx, lavalink):
```

Comment:
This method sends responses, if the play command is invoked via slash commands, while the bot wasn't already in a voice channel, it will notify about joining the channel, but not about the song that just got queued.
The method should take a boolean flag to make it stop responding on success if it comes from here.

Reply:
Tanjun seems to handle this for you, the bot would respond that its joined the channel as well as that song was added to the queue. I have made changes to prevent the double respond at your request though.